zebra-state/src/request.rs

```diff
@@ -849,6 +849,36 @@ pub enum ReadRequest {
     /// * [`ReadResponse::OrchardTree(None)`](crate::ReadResponse::OrchardTree) otherwise.
     OrchardTree(HashOrHeight),
 
+    /// Returns a list of Sapling note commitment subtrees by their indexes,
+    /// starting at `start_index`, and returning up to `limit` subtrees.
+    ///
+    /// Returns
+    ///
+    /// * [`ReadResponse::SaplingSubtree(BTreeMap<_, NoteCommitmentSubtreeData<_>>))`](crate::ReadResponse::SaplingSubtrees)
+    ///
+    /// If there is no subtree at `start_index`, returns an empty list.
+    SaplingSubtrees {
+        /// The index of the first 2^16-leaf subtree to return.
+        start_index: NoteCommitmentSubtreeIndex,
+        /// The maximum number of subtree values to return.
+        limit: Option<NoteCommitmentSubtreeIndex>,
+    },
+
+    /// Returns a list of Orchard note commitment subtrees by their indexes,
+    /// starting at `start_index`, and returning up to `limit` subtrees.
+    ///
+    /// Returns
+    ///
+    /// * [`ReadResponse::OrchardSubtree(BTreeMap<_, NoteCommitmentSubtreeData<_>>))`](crate::ReadResponse::OrchardSubtrees)
+    ///
+    /// If there is no subtree at `start_index`, returns an empty list.
```

Comment:
###### Optional

We can make this more consistent with the rest of the requests:
```suggestion
    /// * An empty list if there is no subtree at `start_index`
    ///
```